kqueen_ui/asset/dynamic/js/validators.js

```diff
@@ -0,0 +1,69 @@
+function validateFileFieldByKeys(element, requiredKeys, parser, validationFailedMessage) {
+  var file = element.files[0],
+      reader = new FileReader(),
+      isValid = false;
+
+  reader.onload = function(event) {
+    var fileText = event.target.result,
+        data = 'error',
+        message = 'Validation pending';
+    try {
+      data = parser(fileText);
+    } catch(err) {
+      message = validationFailedMessage;
+      console.log(err);
+    }
+
+    if (data !== 'error') {
+      var missingKeys = [],
+          countKeys = 0;
+      requiredKeys.forEach(function(key) {
+        if (key in data) {
+          countKeys++;
+        } else {
+          missingKeys.push(key);
+        }
+      });
+      if (countKeys !== requiredKeys.length) {
+        message = 'Missing mandatory keys: ' + missingKeys.join(', ');
+      } else {
+        isValid = true;
+      }
+    }
+    var validator = $("#switchableForm").data().validator;
+    if (isValid) {
+        var submitted = validator.formSubmitted;
+        validator.resetInternals();
+        validator.toHide = validator.errorsFor(element);
+        validator.formSubmitted = submitted;
+        validator.successList.push(element);
+        validator.invalid[element.name] = false;
+        validator.showErrors();
+    } else {
+        var errors = {[element.name]: message};
+        validator.invalid[element.name] = true;
+        validator.showErrors(errors);
+    }
+    validator.stopRequest(element, isValid);
+  };
+  reader.readAsText(file);
+  return 'pending';
+}
+
+function validateJsonFileField(value, element, requiredKeys) {
+  return validateFileFieldByKeys(element, requiredKeys, (d) => JSON.parse(d), 'Not a valid JSON');
+}
+
+function validateYamlFileField(value, element, requiredKeys) {
+  return validateFileFieldByKeys(
+    element, requiredKeys, (d) => jsyaml.safeLoad(d), 'Not a valid YAML');
+}
+
+function validateFieldIsIpCidr(value, element, regexp) {
+  var regexp = new RegExp(
+    '^((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}' +
+    '([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])' +
+    '(/([0-9]|[1-9][0-9]|2[0-4]))?)?$'
+  );
+  return regexp.test(value);
+}
```

Comment:
please, add this validator also 
https://github.com/Mirantis/kqueen-ui/blob/master/kqueen_ui/asset/dynamic/js/kqueen.js#L39